homeassistant/components/ecowitt/sensor.py

```diff
@@ -241,7 +241,12 @@ def _new_sensor(sensor: EcoWittSensor) -> None:
         )
 
         # Hourly rain doesn't reset to fixed hours, it must be measurement state classes
-        if sensor.key in ("hrain_piezomm", "hrain_piezo"):
+        if sensor.key in (
+            "hrain_piezomm",
+            "hrain_piezo",
+            "hourlyrainmm",
+            "hourlyrainin",
```

Comment:
What's the difference between `hrain_piezomm` and `hourlyrainmm`? Why do we need both?

Reply:
Does it mean depending on which rain sensor is connected one, the other, or both will report the hourly rain rate?

Reply:
Sensors report different based on sensor - I don’t know what happens if both are connected. The display unit might have a preference and likely only display one value- but both values might still be sent with the data request/push.

Reply:
OK, I see.

The title of this PR is "Add ecowitt sensors for hourly rain rates in mm and in", but that's not what the PR does.
Can you try to better explain what the PR actually does and also confirm you've verified that your changes work?

Reply:
The rain bucket sensor reports an hourly rate just like the piezo sensor - so its value needs to be handled the same as the piezo rain sensor. If it’s not it reports an error - I’ve been using this code for months so I know it works.

Reply:
Thanks for confirming the changes are tested 👍 

However, can you please clarify what the PR does. The title is "Add ecowitt sensors for hourly rain rates in mm and in" but it doesn't really do that, the PR changes the state class of two of the sensors?

Reply:
I’ll also add someone else changed my original title “Update sensor.py for Hourly Rain Rates mm and in” to “Add ecowitt sensors for hourly rain rates in mm and in”

Reply:
So feel free to make the title to something you prefer.

Reply:
It’s only one sensor - but the sensor can report in inches or millimeters based on a setting so you need to account for both.

Reply:
The way the current code handles the sensor is increasing total - however the value is reset hourly - when the value is reset ha writes a message to the log that’s it’s not an increasing total.

Reply:
"Update sensor.py for Hourly Rain Rates mm and in" is not a good explanation

Please update the PR title to match the implementation and fill in the "Proposed change" section to explain why the change is needed. I can't do it for you because I'm not familiar with the ecowitt integration.

There's an explanation here about why the hourly sensors should be state class measurement: https://github.com/home-assistant/core/pull/90099#issuecomment-1609943505, maybe link to that and explain the non piezo sensors were forgotten in that PR if that's the case?

Once that's done, the PR can be merged 👍

Reply:
Updated - let me know if something else is needed.